packages/file_selector/file_selector_windows/lib/src/dart_file_open_dialog_api.dart

```diff
@@ -0,0 +1,83 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ffi';
+
+import 'package:ffi/ffi.dart';
+import 'package:win32/win32.dart';
+
+/// FileOpenDialogAPI provider, it used to interact with an IFileOpenDialogInstance.
+class FileOpenDialogAPI {
+  /// Sets dialog options.
+  int setOptions(int options, IFileOpenDialog dialog) {
+    return dialog.setOptions(options);
+  }
+
+  /// Returns dialog options.
+  int getOptions(Pointer<Uint32> ptrOptions, IFileOpenDialog dialog) {
+    return dialog.getOptions(ptrOptions);
+  }
+
+  /// Set confirmation button text on dialog.
+  int setOkButtonLabel(String? confirmationText, IFileOpenDialog dialog) {
+    return dialog.setOkButtonLabel(TEXT(confirmationText ?? 'Pick'));
+  }
+
+  /// Sets allowed file type extensions.
+  int setFileTypes(
+      Map<String, String> filterSpecification, IFileOpenDialog dialog) {
+    final Pointer<COMDLG_FILTERSPEC> registerFilterSpecification =
+        calloc<COMDLG_FILTERSPEC>(filterSpecification.length);
```

Comment:
Where is this freed?

Reply:
Thanks for pointing out this, it was a mistake. We modified this implementation to use `arena` instead of a `calloc` statement.